Deploy-CAM.ps1

```diff
@@ -2202,7 +2268,86 @@ else {
     }
 
 
-    # allow interactive input of a bunch of parameters. spCredential is handled in the service principal functions elsewhere in this file
+    # allow interactive input of a bunch of parameters. spCredential is handled in the SP functions elsewhere in this file
+
+    # Check if deploying Root only (ie, DC and vnet already exist)
+    if( -not $deployOverDC ) {
+        $deployOverDC = Read-Host "Do you want to create a new Domain Controller and VNet? Please hit enter to continue with deploying a new domain and vnet or 'no' to maually provide"
+        $deployOverDC = $deployOverDC -like "*n*"
+    }
+
+    $vnetConfig = @{}
+    $vnetConfig.vnetName = $vnetName
+    $vnetConfig.CSsubnetName = $CloudServicesSubnetName
+    $vnetConfig.GWsubnetName = $GatewaySubnetName
+    $vnetConfig.RWsubnetName = $RemoteWorkstationSubnetName
+    if( $deployOverDC ) {
+        # Don't create new DC and vnets
+        # prompt for vnet name, gateway subnet name, remote workstatin subnet name, connection service subnet name
+        do {
+            if ( -not $vnetConfig.vnetName ) {
+                $vnetConfig.vnetName = Read-Host "Provide Connection Service Vnet Name"
+            }
+            if ( -not (Find-AzureRmResource -ResourceGroupNameEquals $rgName -ResourceType "Microsoft.Network/virtualNetworks" -ResourceNameEquals $vnetConfig.vnetName) ) {
```

Comment:
I'm not sure we want to enforce where the vnet lives in resource group land - I figured we'd allow then to just submit the whole vnetID and it could be in any resource group. Also not sure if we're going to get into 'access rights' issues then about verifying that it exists. Hmm- not sure what the best plan is here.

Reply:
My thinking (and I may be wrong) was that the only part of their infrastructure that we would need to hook into is the Vnet and subnets since the DC and storage accounts and whatever can be in separate resource groups or even on premise or whatever so it made sense to me to just assume the root group we want to overlay would be the one with the vnet since it's the only really mandatory thing. I can change it no problem, but I thought this was cleaner and easier since they don't need to provide that super long string.